Tests/Fixtures/Generated/Strings/flat-swift3-context-multiple.swift

```diff
@@ -53,6 +53,8 @@ internal enum L10n {
     internal static let singleline = L10n.tr("LocMultiline", "SINGLELINE")
     /// another single line
     internal static let singleline2 = L10n.tr("LocMultiline", "SINGLELINE2")
+    /// Ceci n'est pas une pipe.
+    internal static let endingWith = L10n.tr("LocMultiline", "ending.with.")
```

Comment:
Well, what should the result be here? The only reason I wrote the logic to keep trailing dots, was to avoid potential collisions for users. But I wouldn't keep all in-between dots as well.

Reply:
For reference, this:
`"..some..dots.and..empty..components.." = "Veni, vidi, vici";`

Produces this (with this PR):
```swift
internal enum Some {
  internal enum Dots {
    internal enum And {
      internal enum Empty {
        /// Veni, vidi, vici
        internal static let components = L10n.tr("LocMultiline", "..some..dots.and..empty..components..")
      }
    }
  }
}
```

Reply:
To me the result should be the same as of the key was `some.dots.and.empty.components` so 5 levels whose name don't contain a dot nor _ in it?

Reply:
TBH, it might be better to just filter out all empty components, instead of having some weirdly complicated logic about keeping some, but dropping other `.`.

Reply:
But my point was: whatever we decide the result of this should be, we should still need to write tests for those tricky cases to (1) explicit be sure not to have regressions on such special cases (2) to be sure the code won't crash similarly to before this PR but for the other tricky cases

Reply:
I'll drop the "keep the trailing dots" logic, and see if I can refactor the keystructure code a bit

Reply:
> TBH, it might be better to just filter out all empty components, instead of having some weirdly complicated logic about keeping some, but dropping other `.`.

https://github.com/SwiftGen/SwiftGen/pull/531#discussion_r221835071

> I wonder if for `levelName` we shouldn't simply always operate over `keyStructure` (so the components list… with empty components always filtered out), which could both simplify the logic of the code and of the rules of what's generated

😜